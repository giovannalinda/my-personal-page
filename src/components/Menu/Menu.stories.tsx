import { Meta, Story } from '@storybook/react'
import { Menu } from './Menu'

export default {
  title: 'Components/Menu',
  component: Menu,
} as Meta

export const Default: Story = () => <Menu />
