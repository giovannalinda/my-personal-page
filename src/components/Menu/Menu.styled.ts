import styled from 'styled-components'
import { theme } from '~/styles'

export const Container = styled.nav``

export const List = styled.ul`
  padding: ${theme.space[5]};
  margin: 28rem 0 0 0;
  overflow-y: scroll;
`

export const ListItem = styled.li`
  cursor: pointer;
  font-size: ${theme.font.sizes.large};
  padding-bottom: ${theme.space[5]};
  text-align: left;
  transition: ${theme.transitions.slow};

  a {
    color: ${theme.colors.white};

    &:hover {
      color: ${theme.colors.gray400};
    }
  }
`
