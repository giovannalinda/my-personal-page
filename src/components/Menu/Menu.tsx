import Link from 'next/link'
import * as S from './Menu.styled'

export function Menu() {
  return (
    <S.Container>
      <S.List>
        <S.ListItem>
          <Link href='/'>IN√çCIO</Link>
        </S.ListItem>
        <S.ListItem>
          <Link href='/about-me'>QUEM SOU</Link>
        </S.ListItem>
        <S.ListItem>
          <Link href='/contact'>CONTATO</Link>
        </S.ListItem>
      </S.List>
    </S.Container>
  )
}
