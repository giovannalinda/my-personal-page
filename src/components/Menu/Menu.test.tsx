import { render } from '~/__helpers__/app-tests'
import { Menu } from './Menu'

describe('Menu', () => {
  it('matches snapshot', () => {
    const { container } = render(<Menu />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
