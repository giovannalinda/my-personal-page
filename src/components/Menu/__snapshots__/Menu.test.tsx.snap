// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu matches snapshot 1`] = `
<nav
  class="sc-bczRLJ"
>
  <ul
    class="sc-gsnTZi gGxZCf"
  >
    <li
      class="sc-dkzDqf cNCoYv"
    >
      <a
        href="/"
      >
        IN√çCIO
      </a>
    </li>
    <li
      class="sc-dkzDqf cNCoYv"
    >
      <a
        href="/about-me"
      >
        QUEM SOU
      </a>
    </li>
    <li
      class="sc-dkzDqf cNCoYv"
    >
      <a
        href="/contact"
      >
        CONTATO
      </a>
    </li>
  </ul>
</nav>
`;
